pkg/bwagreement/config.go

```diff
@@ -31,7 +32,7 @@ func (c Config) Run(ctx context.Context, server *provider.Provider) (err error)
 
 	ns, err := NewServer(c.DatabaseDriver, c.DatabaseURL, zap.L(), k)
 	if err != nil {
-		return err
+		return fmt.Errorf("Can't connect to the database for bwagreement: %s", err)
```

Comment:
I wish the rest of the packages had errors that were human readable.

Reply:
Ah, wait I know how to do it more cleanly. Let's move creating of the Database into Config.Run. Then server takes database as a parameter (rather than the param). And finally add the reconnect inside Config.Run, rather than making it part of `NewDBManager`.